sim/global-types.ts

```diff
@@ -894,17 +992,19 @@ interface Move extends Readonly<BasicEffect & MoveData> {
 	readonly effectType: 'Move'
 }
 
-type MoveHitData = {[targetSlotid: string]: {
-	/** Did this move crit against the target? */
-	crit: boolean;
-	/** The type effectiveness of this move against the target */
-	typeMod: number;
-	/**
-	 * Is this move a Z-Move that broke the target's protection?
-	 * (does 0.25x regular damage)
-	 */
-	zBrokeProtect: boolean;
-}}
+interface MoveHitData {
+	[targetSlotid: string]: {
```

Comment:
Does eslint enforce this change? Is there a way to prevent that?

Reply:
Yes, https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-definitions.md.

I think it's a good rule to have: interfaces for objects, and types for anything else.